### PR TITLE
chore: freeze and update dependencies

### PR DESCRIPTION
**Pull request by tan-zx**
## Problem

Freeze dependencies in shared

Update dependencies flagged by Snyk; closes:
- https://github.com/opengovsg/postmangovsg/pull/1460
- https://github.com/opengovsg/postmangovsg/pull/1462
- https://github.com/opengovsg/postmangovsg/pull/1463
- https://github.com/opengovsg/postmangovsg/pull/1464
- https://github.com/opengovsg/postmangovsg/pull/1466
- https://github.com/opengovsg/postmangovsg/pull/1581
- https://github.com/opengovsg/postmangovsg/pull/1580
- https://github.com/opengovsg/postmangovsg/pull/1561
- https://github.com/opengovsg/postmangovsg/pull/1480
- https://github.com/opengovsg/postmangovsg/pull/1456
- https://github.com/opengovsg/postmangovsg/pull/1443
- https://github.com/opengovsg/postmangovsg/pull/1444
- https://github.com/opengovsg/postmangovsg/pull/1488
- https://github.com/opengovsg/postmangovsg/pull/1591
- https://github.com/opengovsg/postmangovsg/pull/1592
- https://github.com/opengovsg/postmangovsg/pull/1593
- https://github.com/opengovsg/postmangovsg/pull/1594
- https://github.com/opengovsg/postmangovsg/pull/1595

When merged, will close all these PRs.

### CODE REVIEW

**Comment by tan-zx** [comment]
the backend test is failing because of sth wrong with Docker — should re-run the job tmr and it should work

**Comment by stanleynguyen** [comment]
Is there a reason why we're not pinning packages in lambda functions?

**Comment by tan-zx** [comment]
> Is there a reason why we're not pinning packages in lambda functions?

Have done this in latest commit. Have pushed to staging to see if everything works as expected.

File: shared/package.json [hunk 1]
```diff
@@ -25,30 +25,30 @@
...
     "dd-trace": "2.9.1",
-    "lodash": "^4.17.21",
-    "mustache": "^4.2.0",
-    "rimraf": "^3.0.2",
-    "typescript": "^4.6.3",
-    "xss": "^1.0.9"
+    "lodash": "4.17.21",
+    "mustache": "4.2.0",
+    "rimraf": "3.0.2",
+    "typescript": "4.2.3",
```
Comment (stanleynguyen) [inline shared/package.json:50]: Incorrect version. Should be `4.7.3`

Comment (tan-zx) [reply shared/package.json]: bumped

**Review by stanleynguyen — APPROVED** [review]

**Pull request merged**